index.js

```diff
@@ -119,243 +78,174 @@ class CssWriter {
 	}
 }
 
-module.exports = function svelte(options = {}) {
+/** @returns {import('rollup').Plugin} */
+module.exports = function (options = {}) {
+	const extensions = options.extensions || ['.svelte'];
 	const filter = createFilter(options.include, options.exclude);
 
-	const extensions = options.extensions || ['.html', '.svelte'];
-
-	const fixed_options = {};
-
-	Object.keys(options).forEach(key => {
-		// add all options except include, exclude, extensions, and shared
-		if (pluginOptions[key]) return;
-		fixed_options[key] = options[key];
-	});
-
-	if (major_version >= 3) {
-		fixed_options.format = 'esm';
-		fixed_options.sveltePath = options.sveltePath || 'svelte';
-	} else {
-		fixed_options.format = 'es';
-		fixed_options.shared = require.resolve(options.shared || 'svelte/shared.js');
-	}
+	/** @type {import('svelte/types/compiler/interfaces').ModuleFormat} */
+	const format = 'esm', config = { format };
```

Comment:
I wonder if eslint would allow this. We should probably set that up at some point too

Reply:
We already have ESLint running. It's fine with  it. TBH I only did this because this irks me:

```js
/** @type {import('svelte/types/compiler/interfaces').ModuleFormat} */
const format = 'esm';
const config = { format };
```

:neckbeard:

Reply:
Why isn’t this just

```js
const config = { format: 'esm' };
```

Reply:
Because intellisense complains about it at the `compiler(code, { ...config, })` spot, saying that `format: string` does not align with `format: ModuleFormat` type.

There's no technical reason or anything – it's just so that there are no red squiggles on the page